### PR TITLE
Add recipe for libgd

### PR DESCRIPTION
**Pull request by datalogics-aarroyo**
Specify library name and version:  **libgd/2.2.5**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libgd/all/test_package/example.cpp [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#include "gd.h"
+
+int main() {
+    gdImagePtr im;
+    im = gdImageCreate(10, 10);
+    gdImageDestroy (im);
+}
```
Comment (datalogics-robb) [inline recipes/libgd/all/test_package/example.cpp:7]: Undefined Behavior - you must return a value from a function declared to return a value, even main.

File: recipes/libgd/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"), "CMAKE_MINIMUM_REQUIRED(VERSION 2.6 FATAL_ERROR)", '''cmake_minimum_required (VERSION 3.6)
+PROJECT(GD C)
+include(${CMAKE_BINARY_DIR}/../conanbuildinfo.cmake)
+conan_basic_setup()''')
+
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"), 'PROJECT(GD)', '# moved: PROJECT(GD)')
+        
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"), 'if(NOT MINGW AND MSVC_VERSION GREATER 1399)', 'if (BUILD_STATIC_LIBS AND WIN32 AND NOT MINGW AND NOT MSYS)')
+
+
```
Comment (datalogics-robb) [inline recipes/libgd/all/conanfile.py:37]: I see a couple of things PEP8 might not like - you can pip install autopep8 and run autopep8 on the source file to fix them automatically.

File: recipes/libgd/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+class LibgdConan(ConanFile):
+    name = "libgd"
+    license = "https://github.com/libgd/libgd/blob/master/COPYING"
+    url = "https://github.com/libgd/libgd"
+    description = "GD is an open source code library for the dynamic creation of images by programmers."
+    topics = ("images", "graphics")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False]}
```
Comment (datalogics-robb) [inline recipes/libgd/all/conanfile.py:12]: conan-center-index has a requirement to handle fPIC as an option - so you should add that here too.
Ideally we can get this merged into the main fork, and depending on how popular this library is, others will maintain the package for us.

File: recipes/libgd/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename('libgd-' + self.version, self._source_subfolder)
+        tools.replace_in_file(
+            os.path.join(
+                self._source_subfolder,
+                "CMakeLists.txt"),
+            "CMAKE_MINIMUM_REQUIRED(VERSION 2.6 FATAL_ERROR)",
```
Comment (datalogics-robb) [inline recipes/libgd/all/conanfile.py:37]: It's getting complicated enough you might want to do this with a patch instead - but it's your call.

Comment (datalogics-aarroyo) [reply recipes/libgd/all/conanfile.py:37]: ill make a patch before I submit to conan-center-io, but I will use the current recipe to continue my work on pdf2img.

**Pull request merged**